docs/CLI/6_reference_types.md

```diff
@@ -0,0 +1,185 @@
+
+- Using `oras discover`, view the graph of artifacts now stored in the registry
+
+  ```bash
+  oras discover -o tree $IMAGE
+  ```
+- The output shows the beginning of a graph of artifacts, where the signature is viewed as a child of the $IMAGE
+
+  ```output
+  localhost:5000/net-monitor:v1
+  └── signature/example
+      └── sha256:1b6308bc4a2dd8933e9f66ff5bbc47e685516e5378208b46c58dc...
+  ```
+
+## Creating Deep Graphs of Artifacts
+
+The ORAS Artifacts specification enables deep graphs, enabling signed SBoMs and other artifact types.
+
+- Create and push a sample Software Bill of Materials to the registry:
+  ```bash
+  echo '{"version": "0.0.0.0", "artifact": "'${IMAGE}'", "contents": "good"}' > sbom.json
+
+  oras push $REGISTRY/$REPO \
+    --artifact-type 'sbom/example' \
+    --subject $IMAGE \
+    ./sbom.json:application/json
+  ```
+- Sign the SBoM
+  ```bash
+  SBOM_DIGEST=$(oras discover -o json \
+                  --artifact-type sbom/example \
+                  $IMAGE | jq -r ".references[0].digest")
+
+  echo '{"artifact": "'$REGISTRY/$REPO/$SBOM_DIGEST'", "signature": "pat hancock"}' > sbom-signature.json
+
+  oras push $REGISTRY/$REPO \
+    --artifact-type 'signature/example' \
+    --subject $REGISTRY/$REPO@$SBOM_DIGEST \
+    ./sbom-signature.json:application/json
+  ```
+- View the graph
+  ```bash
+  oras discover -o tree $IMAGE
+  ```
+- Generates the following output:
+  ```bash
+  localhost:5000/net-monitor:v1
+  ├── signature/example
+  │   └── sha256:49f47c674c0224c72ca646ae0b8b70c14115cf4874f7...
+  └── sbom/example
+      └── sha256:dc737e2b9bb2489aa61f3fc4a90e2ec166bd9685fd56c6f48d...
+          └── signature/example
+              └── sha256:31eb6a50c54df208a09222127a06e9b7afe1dd042771631b175...
+  ```
+- Pull the SBOM
+  ```bash
+  # Get the digest for the SBOM
+  SBOM_DIGEST=$(oras discover -o json \
+                  --artifact-type 'sbom/example' \
+                  $IMAGE | jq -r ".references[0].digest")
+  
+  # Create a clean directory for downloading
+  mkdir ./download
+
+  # Pull the SBOM into the download directory
+  oras pull -a -o ./download $REGISTRY/$REPO@$SBOM_DIGEST
+
+  # View the $IMAGE SBOM
+  cat ./download/sbom.json | jq
+  ```
+
+## Registry Support
+
+The following registries currently support the [ORAS Artifacts Specification][oras-artifacts].
+
+- Please [submit PRs](https://github.com/oras-project/oras-www/pulls) for additional registry support.
+- See [ORAS Artifacts Community](https://github.com/oras-project/artifacts-spec#community) for how to get engaged.
+
+### CNCF Distribution with ORAS Artifacts Support
+
+A reference implementation of the ORAS Artifacts Spec is available at [github.com/oras-project/distribution](https://github.com/oras-project/distribution)
+
+To run distribution locally:
+  ```bash
+  docker run -d -p 5000:5000 ghcr.io/oras-project/registry:v0.0.3-alpha
+  REGISTRY=localhost:5000
```

Comment:
Why do you need this environment variable to run the registry locally?

Reply:
It's used by several of the commands in the script. This section sets up the variables to CNCF Distribution, ACR, and eventually ECR, Docker Hub, others